packages/react-components/src/components/table-output-component.tsx

```diff
@@ -212,23 +219,59 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
         this.setState({
             tableColumns: this.columnArray
         });
+        if (this.columnApi) {
+            const columns = this.columnApi.getAllColumns();
+            const timestampHeader = columns.find(column => column.getColDef().headerName === 'Timestamp ns');
+            if (timestampHeader) {
+                this.timestampCol = timestampHeader.getColDef().field;
+            }
+        }
+
     }
 
     private async handleTimeSelectionChange(range: TimelineChart.TimeGraphRange) {
         const start = Math.trunc(this.props.range.getstart() + range.start);
-        if (this.lastIndex.timestamp === start) {
+        const end = Math.trunc(this.props.range.getstart() + range.end);
+
+        this.startTimeStamp = Math.min(start, end);
+        this.endTimeStamp = Math.max(start, end);
+        const startIndex = await this.fetchTableIndex(start);
+
+        if (this.lastSelectedRange.startTimestamp === start &&
+            this.lastSelectedRange.endTimestamp === end &&
+            startIndex === startIndex) {
             return;
         }
-        const index = await this.fetchTableIndex(start);
-        if (index && this.gridApi) {
-            this.lastIndex = { timestamp: start, index: index };
+
+        if (this.gridApi && startIndex) {
+            this.lastSelectedRange = {
+                startTimestamp: start,
+                endTimestamp: end,
+                startIndex: startIndex
+            };
             this.gridApi.deselectAll();
-            this.gridApi.ensureIndexVisible(index);
-            const node = this.gridApi.getDisplayedRowAtIndex(index);
-            if (node.id) {
-                node.setSelected(true);
+            this.pendingIndex = startIndex;
+            this.gridApi.ensureIndexVisible(startIndex);
+            this.selectRows();
+        }
+    }
+
+    private selectRows(): void {
+        if (this.gridApi) {
+            if (this.startTimeStamp === this.endTimeStamp && this.pendingIndex) {
```

Comment:
When you click or shift-click on the event table, the selection made by the user is not recorded. However it triggers a handleTimeSelectionChange event. There, the startTimeStamp and endTimeStamp is recorded. The pending index is undefined after the row has been selected.

If you then scroll out of range to trigger a model update, and then scroll back, you will see that the selection is now only based on timestamps, and it might have changed from the user selection (this only happens when there are multiple events on the same start/end timestamp).

This means that the selection start/end index should be stored and remembered, at least when it is set by the user clicking on the table or selecting events with (Shift+)arrow up/down.